### ISSUE TITLE
partial word searches like "pneumoni" crash kiwix-serve

### ISSUE THREAD

**Issue by holta** [iiab/iiab#684]
### Expected behavior and actual behavior.

When any recent kiwix-serve (e.g. from 2017-12-27) @tim-moody can crash kiwix-serve repeatedly if not always by entering search term "pneumoni" (without the final 'a' !)

e,g, when searching within wikipedia_en_medicine_2017-10 or wikipedia_en_medicine_2018-01

(Press Enter immediately after typing in "pneumoni", regardless if the autocomplete appears immediately or not!)

Sometimes it claims the ZIM has no FT index (full-text index) which is false.

### Steps to reproduce the problem.

Might be a timing thing...intermittent :/

### Specifications like the version of the IIAB, operating system version, or hardware details.

Tested on Raspbian 2017-09-07.

@tim-moody asks if we should follow MySQL and add a restart:
```# Restart crashed server only, on-failure would also restart, for example, when
# my.cnf contains unknown option
Restart=on-abort
RestartSec=5s
```

**Comment by holta** [comment]
@georgejhunt apologies: on today's community/team call (http://minutes.iiab.io) Tim & the group are hoping you might have some recommendations reviewing this idea!

**Comment by holta** [comment]
> @tim-moody asks if we should follow MySQL and add a restart:
> ```
> # my.cnf contains unknown option
> Restart=on-abort
> RestartSec=5s

@tim-moody might possibly have a PR ready for tmrw's community/team call (http://minutes.iiab.io)

**Comment by holta** [comment]
@tim-moody do you know if this can officially be closed now, as @kelson42 suggests, with PR #691 now merged?

Plz clarify the readiness of your PR #687 too if poss!

**Comment by tim-moody** [comment]
I have not tested this. #687 is independent of whether it works and only becomes unnecessary if kiwix-serve never crashes again. It is ready to merge.

**Comment by holta** [comment]
Appears fixed by #691 "kiwix-tools 2017-12-27 -> 2018-02-26" (when testing against large Wikipedias...)
